src/plot_api/plot_api.js

```diff
@@ -381,10 +382,21 @@ function _doPlot(gd, data, layout, config) {
         // happens outside of marginPushers where all the other automargins are
         // calculated. Would be much better to separate margin calculations from
         // component drawing - see https://github.com/plotly/plotly.js/issues/2704
-        Plots.doAutoMargin,
-        Plots.previousPromises
+        Plots.doAutoMargin
     );
 
+    function saveRangeInitialForInsideTickLabels(gd) {
```

Comment:
I'm not clear on why this was added? How does it relate to the chart2music? @ayjayt

Reply:
Actually, looking at the git blame it looks like this function was added in 2020 by @archmoj? I'm not sure why it's even showing up in this diff

Reply:
Could you please point to the commit that added this?

Reply:
If we don't have this on master, this function should be removed. It might be the result of a merge conflict going wrong.

Reply:
There are two commits for this function: 
* https://github.com/plotly/plotly.js/commit/5ca95d3794159ace3eb623ea1a7f62a3f6971106
* https://github.com/plotly/plotly.js/commit/445eee4ee21cadef37981e57db63619ef6d12b66

Reply:
This function is not relant to this PR.
Please remove it.